scripts/tests/format.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
```

Comment:
I'm wondering if this script is overkill. Here's what I mean:

- When I'm working on a PR, I can just run `cargo fmt` and then `taplo fmt` from the project directory. I don't care what the check says, I'm just happy when cargo and taplo do their job. Maybe I would like `format.sh` to run both of these commands.
- And when running a workflow, I just want to run `cargo fmt --check` and `taplo fmt --check`. Installing the necessary tools is already handled in previous steps. If they return a non-zero code, the workflow will fail (no need for `check_return_code`, I think). I wouldn't put this in a script, but rather directly in the workflow.

So in both cases, it seems to me like this script is not required. Did you have a particular purpose in mind for this?

Reply:
format script can have other formatting related checks for example, checking for spelling mistakes, formatting .TS files etc.
Also personally I am not very much inclined towards scripts and to me only fuzz.sh seems value able all other scripts are overkill, clippy.sh has a single line :sweat_smile:

Reply:
Yeah, the `clippy.sh` is my fault. 😅

I guess what I'm asking is: Why isn't `format.sh` the following right now:

```shell
cargo fmt --check
taplo fmt --check
```

Reply:
I think it has just simple functions which lets specify --check when in-place changes are not expected (like CI) and sames script execution with out --check runs commands such that required changes are made.

Reply:
but I am fine with any way.

Reply:
Oh, I see. I think we need check only in CI and in-place (without check) on local copies. So I don't see much use for the separation. But I'll let the second review decide. @sea212 @Chralt98